packages/core/src/utilities/VoxelValue.ts

```diff
@@ -0,0 +1,235 @@
+import type { BoundsIJK, Point3, VolumeScalarData } from '../types';
+
+/**
+ * This is a simple, standard interface to values associated with a voxel.
+ */
+export default class VoxelValue<T> {
+  public modifiedSlices = new Set<number>();
+  public boundsIJK = [
+    [Infinity, -Infinity],
+    [Infinity, -Infinity],
+    [Infinity, -Infinity],
+  ] as BoundsIJK;
+
+  // Provide direct access to the underlying data, if any
+  public scalarData: VolumeScalarData;
+  public map: Map<number, T>;
+  public sourceVoxelValue: VoxelValue<T>;
+  public isInObject: (pointIPS, pointIJK) => boolean;
+
+  points: Set<number>;
+  dimensions: Point3;
+  width: number;
+  frameSize: number;
+  _get: (index: number) => T;
+  _set: (index: number, v: T) => boolean | void;
+
+  constructor(
+    dimensions,
+    _get: (index: number) => T,
+    _set?: (index: number, v: T) => boolean | void
+  ) {
+    this.dimensions = dimensions;
+    this.width = dimensions[0];
+    this.frameSize = this.width * dimensions[1];
+    this._get = _get;
+    this._set = _set;
+  }
+
+  public getIJK = (i, j, k) => {
+    const index = i + j * this.width + k * this.frameSize;
+    return this._get(index);
+  };
+
+  public setIJK = (i: number, j: number, k: number, v) => {
```

Comment:
how about `setValueAtIJK([i,j,k], value)`

Reply:
What about getAt and setAt for naming?  Then:
getAtIndex
getAtIJK
getAt(ijk: point3)
or the last once could even just be get, but getAt as the prefix is ok.

Reply:
I like it

Reply:
getAtIJK
getAtIJKPoint or getAtIJKArray